File: constructor/ctorCallingMethod/baseTypeDerivedObjectOrder.cs
using System;

class Parent
{
	public Parent()
	{
		Console.WriteLine("Parent CTOR");
		Test();
	}

	virtual public void SayHi()
	{
		Console.WriteLine("Parent Said Hi");
	}

	virtual public void Test()
	{
		Console.WriteLine("Parent Test");
	}
}

class Child : Parent
{
	public Child()
	{
		Console.WriteLine("Child CTOR");
		Test();
	}

	override public void SayHi()
	{
		Console.WriteLine("Child Said Hi");
	}

	override public void Test()
	{
		Console.WriteLine("Child Test");
	}
}

namespace testCsharp
{

	class Program
	{
		static void Main(string[] args)
		{
			Parent child = new Child();
			child.SayHi();
		}
	}
}

// Parent CTOR
// Child Test
// Child CTOR
// Child Test
// Child Said Hi

File: constructor/ctorCallingMethod/baseTypeDerivedObjectOrder.cpp
#include <iostream>
#include <string>

using namespace std;

class Parent
{
public:
	Parent()
	{
		cout << "========= Called by parent constructor =========" << endl;
		cout << "Parent CTOR" << endl;
		NormalOperation();
		VirtualOperation();
		// OperationOverriddenByChild();
	}

	void NormalOperation()
	{
		cout << "Parent calls normal operation" << endl;
	}

	virtual void OperationOverriddenByChild() = 0;

	virtual void VirtualOperation()
	{
		cout << "Parent calls virtual operation" << endl;
	};
};

class Child : public Parent
{
public:
	Child()
	{
		cout << "\n========= Called by child constructor =========" << endl;
		cout << "Child CTOR" << endl;
		NormalOperation();
		VirtualOperation();
		OperationOverriddenByChild();
	}

	void NormalOperation()
	{
		cout << "Child calls normal operation" << endl;
	}

	void OperationOverriddenByChild() override
	{
		cout << "Child calls overridden operation" << endl;
	};

	void VirtualOperation()
	{
		cout << "Child calls virtual operation" << endl;
	};
};

int main()
{
	Parent *child = new Child();
	cout << "\n========= Normal method calls =========" << endl;

	child->NormalOperation();
	child->VirtualOperation();
	child->OperationOverriddenByChild();

	delete child;
	return 0;
};

// ========= Called by parent constructor =========
// Parent CTOR
// Parent calls normal operation
// Parent calls virtual operation

// ========= Called by child constructor =========
// Child CTOR
// Child calls normal operation
// Child calls virtual operation
// Child calls overridden operation

// ========= Normal method calls =========
// Parent calls normal operation
// Child calls virtual operation
// Child calls overridden operation